tests/exampleTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions\Tests;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests for when this package is installed in a vendored directory
+ *
+ * @group gcf-framework
+ */
+class exampleTest extends TestCase
+{
+    private static $containerId;
+    private static $imageId;
+
+    public static function setUpBeforeClass(): void
+    {
+        if ('true' === getenv('TRAVIS')) {
+            self::markTestSkipped('These tests do not pass on travis');
+        }
+
+        $exampleDir = __DIR__ . '/../examples/hello';
+        self::$imageId = 'test-image-' . time();
+
+        $cmd = 'composer update -d ' . $exampleDir;
+
+        passthru($cmd, $output);
+
+        $cmd = sprintf('docker build %s -t %s', $exampleDir, self::$imageId);
+
+        passthru($cmd, $output);
+
+        $cmd = 'docker run -d -p 8080:8080 '
```

Comment:
nit: By adding `--rm` you can avoid the extra steps to extract the container ID and separately remove it.

```suggestion
        $cmd = 'docker run --rm -d -p 8080:8080 '
```

Reply:
genius!

Reply:
I think I still need the container ID in order to stop it though...

Reply:
Oh right. You could add `--name` and provide your own, which may be slightly simpler.

Reply:
I messed around with this option and the tests failed. I believe if you supply `--rm` and `-d` it doesn't work right? Anyway, I'll be skipping this for now since the current solution works.